python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -1244,3 +1355,131 @@ def disjoint(self, other, align=True):
+
+    def _basic_equals(self, other):
+        """Utility method that returns True if any point in the lhs geometry
+        is equal to a point in the rhs geometry."""
+        from cuspatial.core.binops.equals_count import (
+            pairwise_multipoint_equals_count,
+        )
+
+        lhs = _multipoints_from_geometry(self)
+        rhs = _multipoints_from_geometry(other)
+        result = pairwise_multipoint_equals_count(lhs, rhs)
+        return result > 0
+
+    def _basic_equals_all(self, other):
+        """Utility method that returns True if all points in the lhs geometry
+        are equal to points in the rhs geometry."""
+        from cuspatial.core.binops.equals_count import (
+            pairwise_multipoint_equals_count,
+        )
+
+        lhs = _multipoints_from_geometry(self)
+        rhs = _multipoints_from_geometry(other)
+        result = pairwise_multipoint_equals_count(lhs, rhs)
+        sizes = (
+            lhs.multipoints.geometry_offset[1:]
+            - lhs.multipoints.geometry_offset[:-1]
+        )
+        return result == sizes
+
+    def _basic_equals_count(self, other):
+        """Utility method that returns the number of points in the lhs geometry
+        that are equal to a point in the rhs geometry."""
+        from cuspatial.core.binops.equals_count import (
+            pairwise_multipoint_equals_count,
+        )
+
+        lhs = _multipoints_from_geometry(self)
+        rhs = _multipoints_from_geometry(other)
+        result = pairwise_multipoint_equals_count(lhs, rhs)
+        return result
+
+    def _basic_intersects_pli(self, other):
+        """Utility method that returns the original results of
+        `pairwise_linestring_intersection`."""
+        from cuspatial.core.binops.intersection import (
+            pairwise_linestring_intersection,
+        )
+
+        lhs = _linestrings_from_geometry(self)
+        rhs = _linestrings_from_geometry(other)
+        return pairwise_linestring_intersection(lhs, rhs)
+
+    def _basic_intersects_count(self, other):
+        """Utility method that returns the number of points in the lhs geometry
+        that intersect with the rhs geometry."""
+        pli = self._basic_intersects_pli(other)
+        if len(pli[1]) == 0:
+            return _zero_series(len(other))
+        intersections = _points_and_lines_to_multipoints(pli[1], pli[0])
+        sizes = cudf.Series(intersections.sizes)
+        # If the result is degenerate
+        is_degenerate = _multipoints_is_degenerate(intersections)
+        # If all the points in the intersection are in the rhs
+        if len(is_degenerate) > 0:
+            sizes[is_degenerate] = 1
+        return sizes
+
+    def _basic_intersects(self, other):
+        """Utility method that returns True if any point in the lhs geometry
+        intersects with the rhs geometry."""
+        is_sizes = self._basic_intersects_count(other)
+        return is_sizes > 0
+
+    def _basic_intersects_through(self, other):
```

Comment:
Especially here, it suggests that the line passes "through" another object and therefore has at least two intersections. It isn't strictly accurate though as it could touch and exit and touch again and produce the same result. I think I will drop it.